src/main/java/Duke.java

```diff
@@ -1,10 +1,378 @@
+            try {
+                DukeException.validate(command);
+            } catch(DukeException ex) {
+                System.out.println("Sorry, I am unable to do that at the current moment.");
+                System.out.println("Please type 'help' for a list of commands you can give me! :)");
+            }
+
+            // Calling the method based on what the user input is.
+            switch (command) {
+                case "help":
+                    help();
+                    break;
+
+                case "deadline":
+                    deadline();
+                    break;
+
+                case "todo":
+                    todo();
+                    break;
+
+                case "event":
+                    event();
+                    break;
+
+                case "list":
+                    list();
+                    break;
+
+                case "mark":
+                    mark();
+                    break;
+
+                case "unmark":
+                    unmark();
+                    break;
+
+                case "delete":
+                    delete();
+                    break;
+
+                case "bye":
+                    try {
+                        save("list.txt");
+                    } catch(IOException ex) {
+                        System.out.println("Something went wrong: " + ex.getMessage());
+                    }
+                    bye();
+
+                default:
+                    break;
+            }
+        }
+    }
+
+    /**
+     * Lists all the commands the user can give Mitsuki.
+     */
+    public static void help() {
+        System.out.println("Here is a list of commands you can give me:");
+        for (int i = 0; i < commands.size(); i++) {
+            int j = i + 1;
+            System.out.println(j + ". " + commands.get(i));
+        }
+    }
+
+    /**
+     * Adds a deadline item to the user's list.
+     * Uses '/' to separate the deadline from the task.
+     * E.g. 'deadline Return book /By Sunday 2pm' adds the task
+     *      [D][ ] Return book (By Sunday 2pm)
+     *      to the user's list
+     * Throws an exception if empty task description or no deadline is given.
+     */
+    public static void deadline() {
+        String fullString = scan.nextLine();
+
+        String[] tokens = fullString.split("/");
+        try {
+            EmptyTaskException.timedValidate(tokens);
+            MissingDeadlineException.validate(tokens);
+        } catch(EmptyTaskException ex) {
+            System.out.println("Please give a description for your deadline item. Try again!");
+            return;
+        } catch(MissingDeadlineException ex) {
+            System.out.println("Please give a deadline for your deadline item. Try again! XD");
+            System.out.println("E.g. Type 'deadline return book /by Sunday 2pm' to add the task 'return book' " +
+                    "\nwith a deadline of 'Sunday 2pm' to your list.");
+            return;
```

Comment:
I really like how you put in extra work to catch the different errors. Good job! :)